classes/cmake.yaml

```diff
@@ -19,17 +23,20 @@ buildScript: |
     [ -z "${CXXFLAGS:+true}" ] || export CXXFLAGS
 
     # Create a toolchain file if cross-compiling
-    if [[ "${AUTOCONF_BUILD:-unknown}" != "${AUTOCONF_HOST:-${AUTOCONF_BUILD:-unknown}}" ]]; then
-        CMAKE_SYSTEM_NAME=
-        case "$AUTOCONF_HOST" in
-            *-linux-*)
-                CMAKE_SYSTEM_NAME=Linux
-                ;;
-            *)
-                CMAKE_SYSTEM_NAME=Generic
-                ;;
-        esac
-        cat >toolchain.cmake <<EOF
+    if [[ ${CMAKE_TOOLCHAIN_FILE:+true} ]] ; then
+        CMAKE_TOOLCHAIN_FILE="${BOB_TOOL_PATHS[target-toolchain]}/${CMAKE_TOOLCHAIN_FILE}"
+    else
```

Comment:
I would rather not re-introduce this historic mistake. The toolchain itself should not bother about the build systems used downstream. If CMake needs a toolchain file then this should be handled by the CMake class. That's the same as for the meson class that also needs something similar.

Reply:
Okay. Understand, I'll fix this.